FPGA: Update the include folder readme (#861)
oneapi-src/oneAPI-samples: DirectProgramming/DPC++FPGA/include/README.md

This directory contains utility header libraries optimized for FPGA DPC++ designs. Examples of their usage can be found in ReferenceDesigns and Tutorials.

## Available Header Libraries

### Utilities

| Filename       | Description
---              |---
| constexpr_math.hpp | Defines utilities for statically computing math functions such as Log2.

> | pipe_utils.hpp | Defines utilities for working with pipes, such as PipeArray.

Reviewer: Since you're monkeying with this file anyway, do you mind getting rid of 'Defines . . .' everywhere?  I've given a few examples.

```suggestion
| pipe_utils.hpp | Utility classes for working with pipes, such as PipeArray.
```

| tuple.hpp | Defines a template to implement tuples.
| unrolled_loop.hpp | Defines a templated implementation of unrolled loops.

### Linear algebra

| Filename       | Description
---              |---

> | streaming_qrd.hpp | Defines a functor that implements the QR decomposition of matrices.

Reviewer: ```suggestion
| streaming_qrd.hpp | QR decomposition of matrices with pipe interfaces.
```

> | streaming_qri.hpp | Defines a functor that implements the QR-based inversion of matrices.

Reviewer: ```suggestion
| streaming_qri.hpp | QR-based inversion of matrices with pipe interfaces.
```
